Source/WebCore/xml/XPathStep.cpp

```diff
@@ -348,7 +354,13 @@ void Step::nodesInAxis(Node& context, NodeSet& nodes) const
 
             // Avoid lazily creating attribute nodes for attributes that we do not need anyway.
             if (m_nodeTest.m_kind == NodeTest::NameTest && m_nodeTest.m_data != starAtom()) {
-                auto attr = contextElement.getAttributeNodeNS(m_nodeTest.m_namespaceURI, m_nodeTest.m_data);
+                RefPtr<Attr> attr;
+                // We need this branch because getAttributeNodeNS() doesn't do
+                // ignore-case matching even for an HTML element in an HTML document.
+                if (m_nodeTest.m_namespaceURI.isNull())
+                    attr = contextElement.getAttributeNode(m_nodeTest.m_data);
```

Comment:
Wouldn't this do the wrong thing outside of an HTML document or for non-HTML elements?

Reply:
I don't think there is an issue? `getAttributeNode()` will only ignore the case if this returned true:
```
inline bool shouldIgnoreAttributeCase(const Element& element)
{
    return element.isHTMLElement() && element.document().isHTMLDocument();
}
```

Reply:
Ah right, of course!

Reply:
I was thinking there might be a difference in matching with qualified name vs local name, but it looks like you cannot tie a prefix to the null namespace, so it should be all good.